pallets/automation-price/src/lib.rs

```diff
@@ -399,6 +399,14 @@ pub mod pallet {
 	/// Many fields on this struct is optinal to support multiple error condition
 	#[derive(Debug, Encode, Eq, PartialEq, Decode, TypeInfo, Clone)]
 	pub enum TaskCondition {
+		TargetPriceMatched {
+			// The target price the task set to
+			target_price: u128,
+			trigger_function: Vec<u8>,
```

Comment:
Okay, I think there might be confusion in this `condition` paramter, , possibly due to its ambiguity. The goal of the event is not to re-iterate the trigger condition itself, but to provide real-time value of the circumstances that triggered a task in the code. For example, the TaskTriggered event should look like

`TaskTriggered {who, taskId, condition: condition_struct }`
in which, the `condition_struct` mirrors the same structure as `{destination, exchange, asset1, asset2, amount}` with `amount` being the current value on chain, as a proof of trigger.

Keep in mind the definition of condition_struct is likely to change, so we should use the definition as much as possible, rather than deconstructing it, to faciliate future updates.

Reply:
Okay, I think we can omit the task details from the event, since it’s trivial compared to the real-time on-chain price. The reason is because task detais do not change after creation, but price varies across blocks. We almost always need to leverage the Insights service to read the task details from the past.

Reply:
In other words, the task details already have on-chain proof and can be retrieved from Insights, but the triggering price of this moment is not recorded yet.